numba/cuda/tests/cudapy/test_compiler.py

```diff
@@ -43,6 +43,19 @@ def add(x, y):
         # Inferred return type as expected?
         self.assertEqual(resty, float32)
 
+        # Check that function's output matches signature
+        sig_int32 = int32(int32, int32)
+        ptx, resty = compile_ptx(add, sig_int32, device=True)
+        self.assertEqual(resty, int32)
+
+        sig_int16 = int16(int16, int16)
+        ptx, resty = compile_ptx(add, sig_int16, device=True)
+        self.assertEqual(resty, int16)
+
+        sig_string = "uint32(uint32, uint32)"
```

Comment:
Might be worth adding a comment to explicitly point out that this is also testing with a string-based signature in contrast to the signature objects above.